lib/validate.js

```diff
@@ -4,16 +4,36 @@ var assert = require('assert')
   , async = require('async')
   , isObject = require('is-object')
   , getLogger = require('./log')
-  , Joi = require('joi');
+  , Joi = require('joi')
+  , _ = require('lodash');
+
+/**
+* Perform Joi validation.
+* @param {Object} joiSchema - Joi schema
+* @param {Object} requestData -  either req.query or req.body
+* @param {Object} validateOptions - options to pass to Joi.validate
+* @return {Function} callback
+*/
+function doValidate(joiSchema, requestData, validateOpts, callback) {
+  Joi.validate(requestData, joiSchema, validateOpts, function(err, result) {
```

Comment:
I think maybe we can keep this as just _Joi.validate(requestData, joiSchema, validateOpts, callback)_